### ISSUE TITLE
[3.0.0-beta.6]  fileObj._id on upload end ?

### ISSUE THREAD

**Issue by ddaydd** [veliovgroup/Meteor-Files#889]
Hello, 

with the 3.0.0-beta.6 version, how to retrieve the id?

```
      upload.on('end', function (error, fileObj) {
          console.log(fileObj._id); // undefined
    }
```

Thanks.

**Comment by dr-dimitru** [comment]
@bratelefant @jankapunkt @harryadel do we have bug in beta/rc release?

**Comment by dr-dimitru** [comment]
@ddaydd have you found a solution?

**Comment by ddaydd** [comment]
no, and I saw that findOne became findOneAsync but no insertAsync?

**Comment by dr-dimitru** [comment]
@ddaydd `.insert` is client only and only mimics `.insert` method or mongodb driver

**Comment by ddaydd** [comment]
thank you for motivating me...
I found the _id

```
...
let fileId;
upload.on('start', function () {
  fileId = this.config.fileId
});
...
```

**Comment by dr-dimitru** [comment]
@ddaydd I'm sure it's available via `this` inside `end` event as well

**Comment by ddaydd** [comment]
Yes, i confirm, but I leave it open, this must at least be changed or documented. thanks.

**Comment by donstephan** [comment]
I'm seeing this on the client in the `upload.on('end')` callback. The resulting `fileObj` doesn't include a `_id` like it used to. Anyone else experiencing this?

**Comment by donstephan** [comment]
Seems you need to access the `fileId` through the `uploadInstance` config element in `ostrio:files@3.0.0-beta.6`

For example:
```
var uploadInstance = Images.insert({
  file: file,
  chunkSize: 'dynamic'
}, false);

uploadInstance.on('end', function () {
  console.log('File _id: ', uploadInstance.config.fileId);
});

uploadInstance.start();
```

**Comment by dr-dimitru** [comment]
@donstephan 

1. Note: Do not rely on `.fileId` as to `_id` of the file as they may differ for obfuscation purposes (built-into this package)
2. Have you checked file's details on `this` context?

**Comment by donstephan** [comment]
In the example I shared above there's no `this` context relevant to the file upload. This is strictly client side. The resulting `fileObj` doesn't have any `_id` on it like it used to.

To clarify, even though I'm listening to the `'end'` event, the `fileId` in the `uploadInstance` potentially isn't the resulting `_id` of the file per obfuscation (which is fair).

Is this related to this comment in the 3.0 PR? https://github.com/veliovgroup/Meteor-Files/pull/884#discussion_r1612953210

I can share a test if needed.

**Comment by bratelefant** [comment]
Ok, when I eventually started moving my full app to 3.0.1 I also can confirm this is a breaking change. I don't get a proper fileRef in the "uploaded" event listener callback; worked fine in 2.16. Trying to figure this out...

**Comment by timsun28** [comment]
I'm having the same issue after updating to meteor 3.0 and using v3.0.0-beta.6.
I also found that the onUploaded hook doesn't work properly anymore. It doesn't return an error/fileObj and this is undefined and it also runs instantly after submitting. I'm not sure if this is intended, as it isn't very clear what is new in v3.

Running the following code: 
```
const upload = ElementPhotos.insert(photoObject, false);
upload.on("end", function () {
    console.log({ thisInstance: this });
    console.log("File _id: ", upload.config.fileId);
});
```

The "this" instance contains some details about the file upload and the upload.config.fileId is a random string, but as dr-dimitru pointed out, this shouldn't be used for the required usage.

**Comment by bratelefant** [comment]
Ok just pushed https://github.com/veliovgroup/Meteor-Files/pull/884/commits/d57374198218b178f4a88800fdf8a0dd8c4cc7f9 
This should return the fileObj after it was inserted in the db on "end" and "uploaded" events on the client. Feel free to test and comment

**Comment by timsun28** [comment]
Just installed the package locally with your latest changes and can confirm that this fixed the issue. It now returns the expected object in both the end callback and the onUploaded that includes the _id of the newly inserted file/image and more.

Thank you for your support and hard work on upgrading this package!

**Comment by hluz** [comment]
> Ok just pushed [d573741](https://github.com/veliovgroup/Meteor-Files/commit/d57374198218b178f4a88800fdf8a0dd8c4cc7f9) This should return the fileObj after it was inserted in the db on "end" and "uploaded" events on the client. Feel free to test and comment

Any progress on this push?

**Comment by dr-dimitru** [comment]
@bratelefant @timsun28 @hluz @ddaydd `v3.0.0-beta.7` is out, please give it a try

**Comment by dr-dimitru** [comment]
`ostrio:files@3.0.0-rc.2` is out and ready for tests, give it a try and lmk wdyt

**Comment by hluz** [comment]
> [@bratelefant](https://github.com/bratelefant) [@timsun28](https://github.com/timsun28) [@hluz](https://github.com/hluz) [@ddaydd](https://github.com/ddaydd) `v3.0.0-beta.7` is out, please give it a try

Thanks! Will get to it early next week.

**Comment by dr-dimitru** [comment]
@hluz make sure you are testing the lasers release candidate version

**Comment by hluz** [comment]
Testing with RC.3. So far so good. Only changes were related to new API (writeAsync and unlinkAsync). Thanks!

**Comment by dr-dimitru** [comment]
@hluz awesome! Thank you giving it a try

**Comment by ddaydd** [comment]
Thanks, it's OK for me with the RC.3, can we close this issue?

**Comment by dr-dimitru** [comment]
@ddaydd Great, thank you for testing
